File: raw/raw.js
'use strict';

var gm = require("gm")
  , fs = require("fs")

  , output = fs.createWriteStream("/dev/null")
  , image = "../image.jpg"
  , limit = 2, count = 1000;

(function self() {
  --count;
  --limit;
  gm(image)
    .resize(100, 100)
    .stream("jpg", function(err, stdout, stderr) {
      if (err) {
        throw err;
      }
      stdout.pipe(output);
      stdout.on('end', function () {
        ++limit;
        if (count) {
          self();
        }
      });
    });
  if (limit && count) {
    self();
  }
}());

File: raw/raw.rb
count = 1000
limit = 2

1.upto(count / limit) do
  threads = []

  1.upto(limit) do
    threads << Thread.new do
      IO.popen("/usr/local/bin/gm convert -size 100x100 ../image.jpg -resize 100x100 JPG:-") do |result|
        devnull = File.open("/dev/null", "w")

        while part = result.read(1024)
          devnull.puts(part)
        end
      end
    end
  end

  threads.join
end